app/src/main/java/ch/epfl/sweng/erpa/activities/CreateGameActivity.java

```diff
@@ -50,18 +52,21 @@ public void onFragmentInteraction(Uri uri) {
     }
 
     // When Oneshot or Campaign checkboxes are checked, uncheck the other one
+    @OnClick ({R.id.oneshot, R.id.campaign})
     public void onOneShotOrCampaignSelected(View view) {
-        ButterKnife.bind(this, view);
         findViewById(R.id.layout_num_sessions).setVisibility((view.getId() == R.id.campaign) ? View.VISIBLE : View.GONE);
     }
 
     //call when the user submit a game and check if no requested field is empty
+//    @OnClick(R.id.submit_button)
     public void submitGame(View view) {
-        EditText minPlayer = findViewById(R.id.min_num_player_field);
-        EditText maxPlayer = findViewById(R.id.max_num_player_field);
-        int valueMin = Integer.parseInt(minPlayer.getText().toString());
-        int valueMax = Integer.parseInt(maxPlayer.getText().toString());
-        if (!playerNumberIsValid(valueMin, valueMax)) {
+        EditText valueMin = findViewById(R.id.min_num_player_field);
+        EditText valueMax = findViewById(R.id.max_num_player_field);
+        String min = valueMin.getText().toString();
+        String max = valueMax.getText().toString();
+        int minPlayer = min.isEmpty() ? -1 : Integer.parseInt(min);
+        int maxPlayer = max.isEmpty() ? -1 : Integer.parseInt(max);
```

Comment:
parseInt can throw an exception. For instance if the number is too long, or if the input didn't contain a number. Need a try/catch block. Or better input sanitization (though try/catch would be easier)

Reply:
It can't happens because the EditText that gives the value requires an integer, and has a max number of characters set to 3.